### PR TITLE
Add tests to capture discovery output

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind cleanup
/area test

#### What this PR does / why we need it:

Captures output of discovery API (both individual and aggregated) in /api package for review

#### Which issue(s) this PR fixes:
addresses the "Shouldn’t shortnames be under api review?" aspect of https://github.com/kubernetes/kubernetes/pull/117742#issuecomment-1546411695

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/assign @smarterclayton

### CODE REVIEW

File: hack/update-openapi-spec.sh [hunk 1]
```diff
@@ -96,6 +97,12 @@ if ! kube::util::wait_for_url "https://${API_HOST}:${API_PORT}/healthz" "apiserv
   exit 1
 fi
 
+kube::log::status "Updating aggregated discovery"
+
+rm -fr "${DISCOVERY_ROOT_DIR}"
+mkdir -p "${DISCOVERY_ROOT_DIR}"
+curl -kfsS -H 'Authorization: Bearer dummy_token' -H 'Accept: application/json;g=apidiscovery.k8s.io;v=v2beta1;as=APIGroupDiscoveryList' "https://${API_HOST}:${API_PORT}/apis" | jq -S . > "${DISCOVERY_ROOT_DIR}/aggregated_v1beta1.json"
```
Comment (aojea) [inline hack/update-openapi-spec.sh:104]: why this is v2beta1 and the file v1beta1? is there are typo or are different things?

Comment (liggitt) [reply hack/update-openapi-spec.sh:104]: oops, typo

Comment (liggitt) [reply hack/update-openapi-spec.sh]: fixed

**Comment by dims** [comment]
/approve
/lgtm
/hold

please feel free to remove hold @liggitt .. don't know if you want to land the other one first?

**Comment by liggitt** [comment]
will rebase and unhold once https://github.com/kubernetes/kubernetes/pull/117993 merges

**Comment by liggitt** [comment]
https://github.com/kubernetes/kubernetes/pull/117993 merged, rebased and pushed

**Pull request merged**